### PR TITLE
Fix serviceAccountCreds test and computeEngineCreds test.

### LINKED ISSUE #702: compute_engine_creds and service_account_creds failing
They were added in 926a2c1 and ac4952c, but since 131ba5d the tests look like they have been failing.

@madongfly, do you want to fix it?

```
$ /var/local/git/grpc-java/run-test-client.sh --use_tls=true --server_port=443 --server_host=grpc-test.sandbox.google.com --server_host_override=grpc-test.sandbox.google.com --default_service_account=<snip> --oauth_scope=https://www.googleapis.com/auth/xapi.zoo --test_case=compute_engine_creds
...
Running test compute_engine_creds
Exception in thread "main" java.lang.NullPointerException
	at com.google.common.base.Preconditions.checkNotNull(Preconditions.java:210)
	at io.grpc.auth.ClientAuthInterceptor.<init>(ClientAuthInterceptor.java:67)
	at io.grpc.testing.integration.AbstractTransportTest.computeEngineCreds(AbstractTransportTest.java:748)Shutting down

	at io.grpc.testing.integration.TestServiceClient.runTest(TestServiceClient.java:219)
	at io.grpc.testing.integration.TestServiceClient.run(TestServiceClient.java:192)
	at io.grpc.testing.integration.TestServiceClient.main(TestServiceClient.java:79)
```

```
$ /var/local/git/grpc-java/run-test-client.sh --use_tls=true --server_port=443 --server_host=grpc-test.sandbox.google.com --server_host_override=grpc-test.sandbox.google.com --service_account_key_file=<snip> --oauth_scope=https://www.googleapis.com/auth/xapi.zoo --test_case=service_account_creds
...
Running test service_account_creds
Exception in thread "main" java.lang.NullPointerException
	at com.google.common.base.Preconditions.checkNotNull(Preconditions.java:210)
	at io.grpc.auth.ClientAuthInterceptor.<init>(ClientAuthInterceptor.java:67)
	at io.grpc.testing.integration.AbstractTransportTest.serviceAccountCreds(AbstractTransportTest.java:716)
	at io.grpc.testing.integration.TestServiceClient.runTest(TestServiceClient.java:223)Shutting down

	at io.grpc.testing.integration.TestServiceClient.run(TestServiceClient.java:192)
	at io.grpc.testing.integration.TestServiceClient.main(TestServiceClient.java:79)
```

### PR DESCRIPTION
**Pull request by madongfly**
See #702 for details.

Fixes #702. 

@ejona86, Please take a look, thanks!

### CODE REVIEW

**Comment by ejona86** [comment]
That does mean that we don't shutdown the executor in all cases, but I'm content to let the GC do that in a test.

@madongfly LGTM

**Comment by madongfly** [comment]
@ejona86 the executor is static, which caused the test failure, I reverted the previous change and just create an Executor for the two Creds tests, PTAL, thanks!

File: interop-testing/src/main/java/io/grpc/testing/integration/AbstractTransportTest.java [hunk 1]
```diff
@@ -714,7 +714,8 @@ public void serviceAccountCreds(String jsonKey, InputStream credentialsStream, S
         ServiceAccountCredentials.class.cast(GoogleCredentials.fromStream(credentialsStream));
     credentials = credentials.createScoped(Arrays.<String>asList(authScope));
     TestServiceGrpc.TestServiceBlockingStub stub = blockingStub
-        .withInterceptors(new ClientAuthInterceptor(credentials, testServiceExecutor));
+        .withInterceptors(new ClientAuthInterceptor(credentials,
+            Executors.newScheduledThreadPool(2)));
```
Comment (ejona86) [inline interop-testing/src/main/java/io/grpc/testing/integration/AbstractTransportTest.java]: We can just use `Executor.newSingleThreadExecutor()`.

**Comment by ejona86** [comment]
@madongfly LGTM, but with a small comment.

**Pull request merged**